portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,18 +32,77 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<String> comments = new ArrayList<>();
-    comments.add("Hey !!! Just commenting - Prakash");
-    comments.add("Roses are red - Nirmala");
-    comments.add("Violets are blue - Sushant");
-    comments.add("What I am doing, I have no clue- Prashant");  
-    
+
+    Query query = new Query("Comment");
+    PreparedQuery results = datastore.prepare(query);
+
+    int maxCommentLimit = getCommentLimit(request, "max-comment");
+    List<Entity> comments = results.asList(FetchOptions.Builder.withLimit(maxCommentLimit));
+
     Gson gson = new Gson();
     String json = gson.toJson(comments);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String firstName = getParameter(request, "first-name", "");
+    String lastName = getParameter(request, "last-name", "");
+    String commentText = getParameter(request, "comment", "");
+    Date commentDate = new Date();
+
+    Entity commentEntity = new Entity("Comment");
+
+    commentEntity.setProperty("firstName", firstName);
+    commentEntity.setProperty("lastName", lastName);
+    commentEntity.setProperty("commentText", commentText);
+    commentEntity.setProperty("date", commentDate.toString());
+
+    datastore.put(commentEntity);
+    
+    response.sendRedirect("/index.html#connect");
+  }
+  
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").setKeysOnly();
+
+    PreparedQuery results = datastore.prepare(query);
+    
+    for (Entity entity : results.asIterable()) {
+      datastore.delete(entity.getKey());
+    }
+
+    response.getWriter().println("Successfully deleted all entries");
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+
+  private int getCommentLimit(HttpServletRequest request, String parameter) {
+    int maxComment;
+
+    try {
+      maxComment = Integer.parseInt(request.getParameter(parameter));
+    } catch (NumberFormatException e) {
+      maxComment = Integer.MAX_VALUE;
```

Comment:
Do we want to default the max number of comments to such a large number? Unless we want to be able to handle so many comments?

Reply:
that was just something I could think of for handling ALL option when the client wants to view all comments. I did not set a default max number of comments as I did not know how big the comment section is going to grow.

Reply:
Forsure, I think that is something that's up to you as the designer. You can set that limit to 5 if you think that'd be best for the user or 1 million if that's more appropriate :) as long it's something we thought through

Reply:
I think an alternative to that could be 
https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery.html#countentities
Let me know what you think about this.

Reply:
This sounds reasonable to me!

Reply:
The deprecated method would have been useful. The other method needs a fetchoptions and that requires a max-comment.